File: code_wars/kyu_7/013-Minimize_Sum_Of_Array_Array_Series_#1.py
###### Solution 1
def min_sum(arr: list) -> int:
    result = 0
    sort_arr = sorted(arr)

    for i, item in enumerate(sort_arr[: len(arr) // 2]):
        result += item * sort_arr[-(i + 1)]

    return result


###### Solution 2
def min_sum(arr: list) -> int:
    sort_arr = sorted(arr)
    return sum(
        item * sort_arr[-(i + 1)]
        for i, item in enumerate(sort_arr[: len(sort_arr) // 2])
    )


###### Video Link:
###### Problem Link: https://www.codewars.com/kata/5a523566b3bfa84c2e00010b/train/python

File: code_wars/kyu_7/013-Minimize_Sum_Of_Array_Array_Series_#1.js
// ###### Solution 1
function minSum(arr) {
  const sort_arr = arr.sort((a, b) => a - b);
  let result = 0;

  for (let i = 0; i < arr.length / 2; i++)
    result += sort_arr[i] * sort_arr[arr.length - 1 - i];

  return result;
}

// ###### Solution 2
function minSum(arr) {
  const sort_arr = arr.sort((a, b) => a - b);
  return arr
    .slice(0, arr.length / 2)
    .reduce((a, c, i) => a + c * sort_arr[arr.length - 1 - i], 0);
}

// ###### Video Link:
// ###### Problem Link: https://www.codewars.com/kata/5a523566b3bfa84c2e00010b/train/javascript
